base/gmp.jl

```diff
@@ -35,19 +35,43 @@ else
     error("GMP: cannot determine the type mp_limb_t (__gmp_bits_per_limb == $GMP_BITS_PER_LIMB)")
 end
 
+immutable NotFromPool end
 
 type BigInt <: Integer
     alloc::Cint
     size::Cint
     d::Ptr{Limb}
-    function BigInt()
+    function BigInt(::Type{NotFromPool})
         b = new(zero(Cint), zero(Cint), C_NULL)
         ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
-        finalizer(b, _gmp_clear_func)
+        finalizer(b, poolingfinalizer)
+        b
+    end
+    function BigInt()
+        b::BigInt
+        if length(bigintpool) > 0
+            b = pop!(bigintpool)
+            x= string(b) #########   <------- segfaulting!
+        else
+            b = new(zero(Cint), zero(Cint), C_NULL)
+            ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
+        end
+        finalizer(b, poolingfinalizer)
         return b
     end
 end
 
+const bigintpool = BigInt[]
+const BIGINTPOOL_MAXSIZE = 100000
+function poolingfinalizer(b::BigInt)
+    if length(bigintpool) < BIGINTPOOL_MAXSIZE
+        push!(bigintpool, BigInt(NotFromPool))
```

Comment:
@vtjnash yes it should be and it was in a previous commit. The problem is I'm getting a segfault when using popped bigints so I just experimented here pushing a new Bigint. It didn't help BTW. I suspect some kind of bug with the GC but still not sure. This should work as it is right? I can't build Julia in my computer so I'm doing all experimentation here, this is obviously wip as stated in the PR name. I would love some insightnintonthis problem though...